CMakeLists.txt

```diff
@@ -53,6 +53,8 @@ if(WIN32)
 	if(MSVC)
 		# Minimum requirement is WinXP
 		add_definitions(-D_WIN32_WINNT=0x0501)
+	else()
+		add_definitions(-D_WIN32)
```

Comment:
not necessary, it's already defined by mingw-w64-gcc

Reply:
Better safe than sorry - that code is catch-all clause that covers all Windows compilers except MSVC. I.e. what happens if someone tries the Intel compiler with it?

Reply:
well, ICC targetting windows will define _WIN32. But for instance when under cygwin's GCC, it *should not* be defined because even if it produces a windows .exe, it is in all other aspects a reimplemented Unix which provides the Unix API and not the Windows one (e.g it defines `__unix__`, not `_WIN32`) as it was made to port from unices easily. MinGW-w64's compilers on the other hand are GCC and clang configured to target the Win32 API directly and as such define _WIN32 and not `__unix__`

Reply:
OK and does building with Cygwin actually define the WIN32 variable that is tested before it gets to this point? If it does, then you are possibly right but I would hesitate to add Cygwin specific stuff.